src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -343,4 +351,22 @@ mod test {
             assert_eq!(got, exp);
         }
     }
+
+    #[test]
+    fn test_sin() {
+        let tests = vec![
+            (ScalarFuncSig::Sin, Datum::F64(0.0_f64), 0.0_f64),
+            (ScalarFuncSig::Sin, Datum::F64(PI / 4.0_f64), FRAC_1_SQRT_2),
+            (ScalarFuncSig::Sin, Datum::F64(PI / 2.0_f64), 1.0_f64),
+            (ScalarFuncSig::Sin, Datum::F64(PI), 0.0_f64),
```

Comment:
How about `f64::NAN` and `f64::INFINITY`?

Reply:
You mean test them as input, right?

Reply:
yes, I think we need them.